apps/cowswap-frontend/src/modules/injectedWidget/updaters/InjectedWidgetUpdater.tsx

```diff
@@ -37,18 +37,41 @@ const cacheMessages = (event: MessageEvent) => {
   messagesCache[method] = event.data
 }
 
+;(function initInjectedWidget() {
+  const isInIframe = window.top !== window.self
+
+  const top = window.top
+
+  if (!top || !isInIframe) return
+
   /**
    * To avoid delays, immediately send an activation message and start listening messages
    */
-  ; (function initInjectedWidget() {
-    const isInIframe = window.top !== window.self
+  window.addEventListener('message', cacheMessages)
+  postMessageToWindow(top, WidgetMethodsEmit.ACTIVATE, void 0)
 
-    if (!window.top || !isInIframe) return
+  /**
+   * Intercept window.open and anchor clicks to send a message to the parent window
+   * to handle the opening of deeplinks in the parent window
+   */
+  const originalWinOpen = window.open
 
-    window.addEventListener('message', cacheMessages)
+  window.open = function (...args) {
```

Comment:
Yes, this is monkeypatching.
`web3modal` uses `window.open` for deeplinks and this is the only way to intercept it

Reply:
Maybe I'm not understanding the underlying issue enough, but could it be possible to just intercept `onhashchange` or `onpopstate` or the react router to intercept the deep link?

Reply:
Those events are triggered when something changes in History. But opening depplinks it's not a History change it's technically opening of a new tab